frontend/src/pages/ArtifactList.tsx

```diff
@@ -120,21 +120,21 @@ class ArtifactList extends Page<{}, ArtifactListState> {
   }
 
   private async reload(request: ListRequest): Promise<string> {
-    const { response: res, error: err } = await Apis.getMetadataServicePromiseClient().getArtifacts(
-      new GetArtifactsRequest(),
-    );
-
-    if (err) {
-      // Code === 5 means no record found in backend. This is a temporary workaround.
-      // TODO: remove err.code !== 5 check when backend is fixed.
-      if (err.code !== 5) {
-        this.showPageError(serviceErrorToString(err));
-      }
-      return '';
+    // TODO: Consider making an Api method for returning and caching types
+    if (!this.artifactTypesMap || !this.artifactTypesMap.size) {
+      this.artifactTypesMap = await getArtifactTypes(
+        this.api.metadataStoreService,
+        this.showPageError.bind(this),
+      );
+    }
+    if (!this.state.artifacts!.length) {
+      const artifacts = await this.getArtifacts();
+      this.setState({ artifacts });
+      this.clearBanner();
     }
 
-    const artifacts = (res && res.getArtifactsList()) || [];
-    await this.getRowsFromArtifacts(request, artifacts);
+    // Updates state.rows
+    this.getRowsFromArtifacts(request);
```

Comment:
You should keep the `await` here.
Demo of problematic behavior: https://drive.google.com/file/d/1Rrye-umChfyJug9nm8hCiYznZb8DFRw0/view

When artifact list page loads, it
1. first shows a loading spinner
2. then shows 'no artifacts' text
3. finally shows the artifacts list

Adding await here removes the 2. phase, because loading spinner waits until this promise resolves to disappear.